RockWeb/App_Code/FileUploader.ashx.cs

```diff
@@ -133,54 +133,93 @@ private void ProcessContentFile( HttpContext context, HttpPostedFile uploadedFil
             // validate file type (child FileUploader classes, like ImageUploader, can do additional validation);
             this.ValidateFileType( context, uploadedFile );
 
-            //
-            // Get filename and scrub invalid characters.
-            //
-            var filename = Path.GetFileName( uploadedFile.FileName );
-            filename = Regex.Replace( filename, @"[<>*%&:\\]", string.Empty, RegexOptions.CultureInvariant );
 
-            // get folderPath and construct filePath
-            string relativeFolderPath = context.Request.Form["folderPath"] ?? string.Empty;
-            string relativeFilePath = Path.Combine( relativeFolderPath, filename );
-            string rootFolderParam = context.Request.QueryString["rootFolder"];
+            // NEVER TRUST THE CLIENT!!!!
+            string untrustedFileName = uploadedFile.FileName;
+            string untrustedFolderPath = context.Request.Form["folderPath"] ?? string.Empty;
+            string encryptedRootFolder = context.Request.QueryString["rootFolder"];
 
-            string rootFolder = string.Empty;
+        
+            /* Scrub the file name */
 
-            if ( !string.IsNullOrWhiteSpace( rootFolderParam ) )
+            string scrubedFileName = ScrubFileName( untrustedFileName );
+
+            if( string.IsNullOrWhiteSpace( scrubedFileName ) )
             {
-                // if a rootFolder was specified in the URL, decrypt it (it is encrypted to help prevent direct access to filesystem)
-                rootFolder = Rock.Security.Encryption.DecryptString( rootFolderParam );
+                throw new Rock.Web.FileUploadException( "Invalid File Name", System.Net.HttpStatusCode.BadRequest );
             }
 
-            if ( string.IsNullOrWhiteSpace( rootFolder ) )
+            /* Scrub the folder path */
+            string scrubedFolderPath = ScrubFileName( untrustedFolderPath );
```

Comment:
@Tschrock FYI -- In case you are using this in your system before you get our fix (https://github.com/SparkDevNetwork/Rock/commit/afc1aad6449d0de45c6f9f36e514165f132c9f21), this is probably supposed to be ScrubFile**Path** not _Name_.